datafusion/physical-plan/src/joins/hash_join.rs

```diff
@@ -1406,12 +1403,24 @@ impl HashJoinStream {
                 self.hashes_buffer.resize(batch.num_rows(), 0);
                 create_hashes(&keys_values, &self.random_state, &mut self.hashes_buffer)?;
 
+                let (filtered_batch, filtered_hashes) =
+                    if let Some(dynamic_filter) = &self.dynamic_filter_info {
+                        dynamic_filter.filter_probe_batch(
```

Comment:
Hmm, `filter_probe_batch` shouldn't be added as filter on the `hash_join` but rather as `PhysicalExpr` (returning boolean) to filter the `ParquetExec`

Reply:
> Hmm, `filter_probe_batch` shouldn't be added as filter on the `hash_join` but rather as `PhysicalExpr` (returning boolean) to filter the `ParquetExec`

Sure, I'll refactor that, other than that I found a problem with ParquetExec here https://github.com/apache/datafusion/issues/13298